### PR TITLE
Fix broken CVE reference and update links

### PR DESCRIPTION
**Pull request by wvu**
Prefer HTTPS over HTTP, too.

Closes #7773.

### CODE REVIEW

File: lib/msf/core/module/reference.rb [hunk 1]
```diff
@@ -96,17 +96,17 @@ def initialize(in_ctx_id = 'Unknown', in_ctx_val = '')
     self.ctx_val = in_ctx_val
 
     if (in_ctx_id == 'CVE')
-      self.site = "http://cvedetails.com/cve/#{in_ctx_val}/"
+      self.site = "https://cvedetails.com/cve/CVE-#{in_ctx_val}/"
```
Comment (wvu-r7) [inline lib/msf/core/module/reference.rb]: Link changed to ```CVE-``` prefix. Also supports HTTPS.

File: lib/msf/core/module/reference.rb [hunk 2]
```diff
@@ -96,17 +96,17 @@ def initialize(in_ctx_id = 'Unknown', in_ctx_val = '')
...
     elsif (in_ctx_id == 'CWE')
       self.site = "https://cwe.mitre.org/data/definitions/#{in_ctx_val}.html"
     elsif (in_ctx_id == 'BID')
       self.site = "http://www.securityfocus.com/bid/#{in_ctx_val}"
     elsif (in_ctx_id == 'MSB')
-      self.site = "http://technet.microsoft.com/en-us/security/bulletin/#{in_ctx_val}"
+      self.site = "https://technet.microsoft.com/en-us/library/security/#{in_ctx_val}"
```
Comment (wvu-r7) [inline lib/msf/core/module/reference.rb]: Old link redirects to new. Also supports HTTPS.

File: lib/msf/core/module/reference.rb [hunk 3]
```diff
@@ -96,17 +96,17 @@ def initialize(in_ctx_id = 'Unknown', in_ctx_val = '')
...
     elsif (in_ctx_id == 'EDB')
       self.site = "https://www.exploit-db.com/exploits/#{in_ctx_val}"
     elsif (in_ctx_id == 'US-CERT-VU')
-      self.site = "http://www.kb.cert.org/vuls/id/#{in_ctx_val}"
+      self.site = "https://www.kb.cert.org/vuls/id/#{in_ctx_val}"
```
Comment (wvu-r7) [inline lib/msf/core/module/reference.rb]: Supports HTTPS.

**Comment by tdoan-r7** [comment]
# Release Notes

This fix resolves broken CVE reference and update links.

**Pull request merged**